build.sbt

```diff
@@ -17,6 +17,7 @@
 import java.nio.file.Files
 
 val sparkVersion = "3.2.0"
+val scala212_major_minor = "2.12" // this is used by java projects to make sure we publish only once
```

Comment:
i think it is better use the scalaVersion field in the common settings.. and then extract the major_minor from it. you can see examples elsewhere in this sbt how to use arbitrary scala function to process stuff

Reply:
yup sg

Reply:
i assume that this is not too much work (since we have code else where doing this)... so its good to do it correctly

Reply:
this doesn't seem trivial. `getMajorMinorPatch` is easy. but getting `commonSettings -> scalaVersion` seems hard. none of the following work
```
val (major, minor, _) = getMajorMinorPatch((commonSettings / scalaVersion).value)
val (major, minor, _) = getMajorMinorPatch(commonSettings.scalaVersion)
val (major, minor, _) = getMajorMinorPatch((scalaVersion in commonSettings).value)
val (major, minor, _) = getMajorMinorPatch(scalaVersion in commonSettings)
```

Reply:
i did this instead as a workaround

```
val all_scala_versions = Seq(scala212, scala213)
...
lazy val commonSettings = Seq(
  organization := "io.delta",
  scalaVersion := scala212,
  crossScalaVersions := all_scala_versions,
  fork := true
)
...
val (major, minor, _) = getMajorMinorPatch(all_scala_versions.head)
```

Reply:
i could just as easily use `scala212` instead too

Reply:
yeah.. why not just use that. 
simply define at the top
```
val default_scala_version = scala212
val all_scala_versions = Seq(scala212, scala213)
```

then use `default_scala_version` in `commonSettings`... as well as.. in this java only thing.

Reply:
done